server/src/main/java/org/eclipse/lsp/cobol/core/preprocessor/delegates/copybooks/LevelNumberAdjustingListener.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (c) 2022 Broadcom.
+ * The term "Broadcom" refers to Broadcom Inc. and/or its subsidiaries.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *    Broadcom, Inc. - initial API and implementation
+ *
+ */
+
+package org.eclipse.lsp.cobol.core.preprocessor.delegates.copybooks;
+
+import org.antlr.v4.runtime.BufferedTokenStream;
+import org.antlr.v4.runtime.tree.TerminalNode;
+import org.eclipse.lsp.cobol.core.CobolParserBaseListener;
+import org.eclipse.lsp.cobol.core.preprocessor.delegates.util.TokenUtils;
+
+import java.util.ArrayDeque;
+import java.util.Deque;
+
+import static org.eclipse.lsp.cobol.core.CobolParser.LevelNumberContext;
+
+/** This listener adjusts the variable level numbers of copybooks */
+public class LevelNumberAdjustingListener extends CobolParserBaseListener
+    implements PreprocessorStack {
+  private final Deque<StringBuilder> accumulator = new ArrayDeque<>();
+  private final BufferedTokenStream tokens;
+  private final int copybookNumber;
+  private int difference = Integer.MIN_VALUE;
```

Comment:
It can be `Integer` with null as an initial value.

Reply:
I agree, but I prefer initial value-based logic over null-based.